Matrix.py

```diff
@@ -0,0 +1,120 @@
+class Matrix:
+    # in this program Matrix is a tuple of tuples and the inside tuples must be as long as the major ine is.
+    # data is the Matrix (tuple) received from the user
+    def __init__(self, data):
+        counter = 0
+        if type(data) is tuple:
+            for tup in data:
+                if len(tup) == len(data) and type(tup) is tuple:
+                    counter += 1
+        if counter == len(data):
+            self.data = data
+        else:
+            raise ValueError("Invalid matrix")
+
+    def __iter__(self):
+        return iter(self.data)
+
+    def __getitem__(self, item):
+        return self.data[item]
+
+    def __repr__(self):
+        return f"Matrix: {self.data}"
+
+    # str returns the Matrix in the mathemathic format
+    def __str__(self):
+        matrix_in_lines = ""
+        for i in range(len(self.data)):
+            matrix_in_lines += (str(self.data[i]) + "\n")
+        return matrix_in_lines
+
+    def get_tuple(self):
+        return self.data
+
+    # unity method creates a matrix of zeros with a diagonal of ones (unity is the Matrix's size)
+    @staticmethod
+    def unity(unit):
+        major_list = [[1 if i == index else 0 for i in range(unit)] for index in range(unit)]
+        sub_list = [tuple(lt) for lt in major_list]  # convert list of lists to type Matrix. lt are the inside lists
+        matrix_to_return = Matrix((tuple(sub_list)))
+        return matrix_to_return
+
+    # ones method returns a Matrix full of ones which sizes determined by "unit"
+    @staticmethod
+    def ones(unit):
+        major_list = [([1] * unit) for i in range(unit)]
+        sub_list = [tuple(lt) for lt in major_list]  # convert list of lists to type Matrix. lt are the inside lists
+        matrix_to_return = Matrix((tuple(sub_list)))
+        return matrix_to_return
+
+    def __add__(self, other):
+        if type(other) is Matrix:
+            ans = []
+            for tup in range(len(self.data)):
+                ans.append([])
+                for num in range(len(self.data)):
+                    ans[tup].append(self.data[tup][num] + other[tup][num])
+            sub_list = [tuple(ls) for ls in ans]  # convert ans to type Matrix. ls are the inside lists
+            matrix_to_return = Matrix((tuple(sub_list)))
+            return matrix_to_return
+        else:
+            raise SyntaxError("only matrix's summery")
+
+    def __sub__(self, other):
+        if type(other) is Matrix:
+            ans = []
+            for tup in range(len(self.data)):
+                ans.append([])
+                for num in range(len(self.data)):
+                    ans[tup].append(self.data[tup][num] - other[tup][num])
+            sub_list = [tuple(ls) for ls in ans]  # convert list of lists to type Matrix. ls are the inside lists
+            matrix_to_return = Matrix((tuple(sub_list)))
+            return matrix_to_return
+        else:
+            raise SyntaxError("only matrix's subscription")
```

Comment:
for cleaner code, it is better to check first the negative stuff and then continue with the positive.
for example, it would look better if you had done:
```
    def __sub__(self, other):
        if type(other) is not Matrix:
            raise SyntaxError("only matrix's subscription")
        
        ans = []
        for tup in range(len(self.data)):
            ans.append([])
            for num in range(len(self.data)):
                ans[tup].append(self.data[tup][num] - other[tup][num])
        sub_list = [tuple(ls) for ls in ans]  # convert list of lists to type Matrix. ls are the inside lists
        matrix_to_return = Matrix((tuple(sub_list)))
        return matrix_to_return
```

you have this phenomenon also with the functions: `__add__` `__truediv__` and `__init__`.